ir/expression.def

```diff
@@ -269,7 +269,13 @@ class Mux : Operation_Ternary {
 
 class DefaultExpression : Expression {}
 
-class This : Expression {}  // experimental
+// Two different This should not be equal.
+// That's why we use a hidden id field to distinguish them.
+class This : Expression {
+    int id = nextId++;
+ private:
+    static int nextId;
+}  // experimental
```

Comment:
How does this work if an abstract method has two explicit references to `this` in its body?  The parser will create a new `This` for each of them, but it would seem you need them to all be the same if you want things to work properly?  Do we need an explicit pass that will do that for a given method?